eng/pipelines/templates/steps/analyze.yml

```diff
@@ -0,0 +1,21 @@
+parameters:
+  - name: Toolchain
+    type: string
+    default: stable
+
+steps:
+  - template: /eng/pipelines/templates/steps/use-rust.yml@self
+    parameters:
+      Toolchain: ${{ parameters.Toolchain }}
+
+  - pwsh: |
+      . ./eng/common/scripts/Helpers/CommandInvocation-Helpers.ps1
```

Comment:
We should probably put these into a script instead of inline so folks can run it easier locally.

Reply:
@heaths would prefer devs run the cargo commands directly instead of a script

Reply:
Yes. It's not helpful to onboard to a project and have to learn unique commands for each repo. We have enough controls over some defaults in the most common cases, but I should just be able to run `cargo test` in any Rust repo and expect it to work for most cases. And if there are optional features, I should expect `cargo test --all-features` to just work. It's the responsibility of a PR build to run all the things we want to check. We hope - and, sadly, it seems not to happen too often - devs are doing some due diligence before pushing commits, but that's the role of the PR build: make sure everything works.

I'd like to see most scripts removed, actually. Most just run common commands with some extra flags that our PR should do anyway, and I'd rather see do inline for the same reason: these are supported, non-breaking command lines that should just work in any repo.

Our other languages don't require scripts for the basics like building and testing. Maybe code generation, doc validation, etc.. Sure, those are all custom things anyway. But tools/frameworks like `dotnet`, `go`, `cargo`, `nppm`, etc. are all opinionated and should be consistent for the most part.